src/app/time.service.ts

```diff
@@ -0,0 +1,25 @@
+import {Injectable} from '@angular/core';
+
+import {Observable} from 'rxjs';
+
+import * as moment from 'moment';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TimeService {
+  constructor() {}
+
+  timeOfDay(): string {
+    const hour: number = +moment().format('HH');
+    return hour >= 4 && hour <= 11
+      ? 'morning'
+      : hour >= 12 && hour <= 16
+      ? 'afternoon'
+      : hour >= 17 && hour <= 21
+      ? 'evening'
+      : hour > 21 || hour < 4
+      ? 'night'
+      : 'invalid-hour';
```

Comment:
The nested ternary operators are clever, but I think it's more readable as a few if statements:

```suggestion
    if (hour >= 4 && hour <= 11) return 'morning';
    if (hour >= 12 && hour <= 16) return 'afternoon';
    if (hour >= 17 && hour <= 21) return 'evening';
    if (hour > 21 || hour < 4) return 'night';
    return null;
```

Also - you may want to return `null` if the period is invalid.  This will allow you to more easily type check for an invalid period.

Reply:
alias ValidTimeOfDay = 'morning' | 'afternoon' | 'evening' | 'night';
alias InvalidTimeOfDay = 'invalid-before-morning' | 'invalid-after-evening';
timeOfDay(): ValidTimeOfDay|InvalidTimeOfDay
doThingWithTimeOfDay(time: ValidTimeOfDay): void